test/fortran/utils.f90

```diff
@@ -0,0 +1,47 @@
+module test_utils
+    use, intrinsic :: iso_c_binding
+    implicit none
+
+contains
+
+subroutine assert_int(val, ref_val)
+    integer, intent(in) :: val, ref_val
+    if (val /= ref_val) then
+        print '()'
+        print '(i0, "/=", i0)', val, ref_val
+        error stop
```

Comment:
Does error give you a traceback or equivalent? Would be difficult to debug without that. Hopefully cmake's boilerplate can save us

Reply:
I assume, if the test in C (and python) works, spglib in C is fine. Because the fortran interface is very simple wrapper, if the test in fortran went wrong, most likely the test code is wrong, but the fortran wrapper is probably not wrong. Therefore, this `print` information is used for debugging the test code.

To see the traceback, we may have to compile spgilb with debugging option. Can we set `set(CMAKE_BUILD_TYPE Debug)` when `spglib_WITH_TESTS=on`? Or, is it already so?

Reply:
I've tested this a bit and even with `Debug` flag we don't get a traceback. At the same time though, there are no clean Fortran methods of doing so close to what standard C++, `catch2` or `GoogleTest` (closest I've found is [`fortuno`](https://github.com/aradi/fortuno)), but that doesn't define macros to automatically throw messages either.

But for what it's worth, I say it's good enough since we only need it to detect issues and then we can use a debugger.

> Can we set `set(CMAKE_BUILD_TYPE Debug)` when `spglib_WITH_TESTS=on`

We shouldn't because users would want to run tests as well.